src/handlers/router.rs

```diff
@@ -103,10 +102,6 @@ impl Router {
             "textDocument/completion".to_string(),
             Box::new(CompletionHandler::default()),
         );
-        mapping.insert(
-            "completionItem/resolve".to_string(),
```

Comment:
did we determine if this was indeed used? Or was it effectively dead code? 

What is the purpose of the `completionItem/resolve` method? The spec is a little light on details. Is it just an opportunity for the server to add more details to a completion item? I.e. the initial CompletionItem the server produces can be lightweight then if the user interacts with that item then the client gives the server a chance to add more details?

Reply:
Ah I found the relevant line in the docs:

> If computing full completion items is expensive, servers can additionally provide a handler for the completion item resolve request (‘completionItem/resolve’).

So its a way to compute more expensive completion items if the user actually selects it. 

Same question did we determine if this was actually used?

Reply:
https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion

Reply:
Yeah, @scbrickley and I had a conversation yesterday about this. It's not used at all, because our completion generation doesn't even return multiple pages. It's a no-op for us, but we should respond to the rpc call in the case where it gets sent to us.

Reply:
Restored this handler (for now).